libraries/lib-telemetry/details/CommonHeaders.cpp

```diff
@@ -0,0 +1,139 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file CommonHeaders.cpp
+ @brief Define a class, used to setup the common HTTP headers for the telemetry services.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "CommonHeaders.h"
+
+#include <cstdlib>
+#include <sstream>
+
+#include <wx/platinfo.h>
+#include <wx/intl.h>
+
+#include "lib-string-utils/CodeConversions.h"
+#include "lib-network-manager/Request.h"
+#include "lib-string-utils/UrlEncode.h"
+
+namespace audacity
+{
+namespace telemetry
+{
+namespace details
+{
+namespace
+{
+void GetOSString (std::ostringstream& output, const wxPlatformInfo& platformInfo)
+{
+    const wxOperatingSystemId osID = platformInfo.GetOperatingSystemId();
+
+    if (osID & wxOS_WINDOWS)
+    {
+        output <<
+               "Windows NT " <<
+               platformInfo.GetOSMajorVersion() <<
+               "." <<
+               platformInfo.GetOSMinorVersion() <<
+               "; ";
+
+        if (platformInfo.GetArchitecture() == wxARCH_64)
+            output << "Win64; x64";
+        else
+            output << "Win32; x86";
+    }
+    else if (osID & wxOS_MAC)
+    {
+        output << "Macintosh; ";
+#ifdef __arm64__
+        output << "AppleSilicon ";
+#else
+        output << "Intel ";
+#endif
+        if (platformInfo.GetOSMajorVersion() == 10)
+        {
+            output <<
+                   "Mac OS X 10_" <<
+                   platformInfo.GetOSMinorVersion() <<
+                   "_" <<
+                   platformInfo.GetOSMicroVersion();
+        }
+        else
+        {
+            output <<
+                   "macOS 11_" <<
+                   platformInfo.GetOSMinorVersion();
+        }
+    }
+    else
+    {
+        if (osID & wxOS_UNIX_LINUX)
+            output << "Linux; ";
+        else if (osID & wxOS_UNIX_FREEBSD)
+            output << "FreeBSD; ";
+        else if (osID & wxOS_UNIX_OPENBSD)
+            output << "OpenBSD; ";
+        else
+            output << "Unix; ";
+
+        if (platformInfo.GetArchitecture() == wxARCH_64)
```

Comment:
This doesn't seem correct. `GetArchitecture()` gives you the bit-width of your architecture, not an exact Intel instruction set. This will report for example Audacity on ARM as x86 or x64, giving you bogus telemetry info.